drivers/spi/spi_nrfx_spim.c

```diff
@@ -596,19 +602,14 @@ static int spim_nrfx_pm_action(const struct device *dev,
 
 		ret = pinctrl_apply_state(dev_config->pcfg,
```

Comment:
suspend action shall be wrapped by `IS_ENABLED(CONFIG_PM_DEVICE)` and it cannot be called when PM is not enabled (sleep state in pinctrl does not exist, too).
Maybe similar approach could be used like here? https://github.com/zephyrproject-rtos/zephyr/blob/afce5329f772a4306af699c7a06efddf93cadca0/drivers/pwm/pwm_nrfx.c#L302 (already used in uart_nrfx_uarte.c as well).

Reply:
Can do. And we also need to start suspended with `PINCTRL_STATE_SLEEP` set.

Patch is updated.

Actually, both places are not the part I expected comments about. I'm more worried about changes introduced to spi_context.h and the whole idea that device runtime PM should keep device available while SPI context lock is held.

Also, if this is indeed correct solution, other SPI drivers (spi_ambiq.c, spi_ll_stm32.c and spi_smartbond.c) need to be updated to use same approach. Likely this is best to be handled in a separate issue.

Reply:
aligned to scheme used in PWM and UARTE drivers